src/java/org/apache/cassandra/index/sai/disk/format/Version.java

```diff
@@ -103,6 +105,47 @@ public static Version latest()
         return LATEST;
     }
 
+    /**
+     * Calculates the maximum allowed length for SAI index names to ensure generated filenames
+     * do not exceed the system's filename length limit (defined in SchemaConstants.FILENAME_LENGTH).
```

Comment:
Nit: A link might make things easier for IDEs if the referenced constant changes in the future:
```suggestion
     * do not exceed the system's filename length limit (defined in {@link SchemaConstants#FILENAME_LENGTH)}.
```